scripts/dev/check_format_strings.py

```diff
@@ -54,214 +54,379 @@
-                    2
-                ]
-            )
-        )
-
-    def test_invalid_format_chunk(self):
-        self.assertEqual(
-            1,  # number of errors encountered
-            process_all_format_lines(
-                Path('/dummy/path'),
-                [  # actual file content lines
-                    'line 1',
-                    'line 2',
-                    'format(\"hi{\", varName);'
-                ],
-                [  # lines containing actual format statements
-                    2
-                ]
-            )
-        )
+    for f_path in file_to_check:
+        try:
+            cleaned, _ = remove_cpp_comments(f_path)
+        except UnicodeDecodeError:
+            # Skip files with unexpected encoding
+            continue
+        for fmt_dict in find_and_parse_format_calls(cleaned):
+            num_errors += check_format_statement(f_path, fmt_dict)
+    return 1 if num_errors > 0 else 0
+
+
+# ========================= File discovery & CLI =========================
+def get_sorted_file_list(search_path: Path) -> List[Path]:
+    out: List[Path] = []
+    for root, _, files in os.walk(search_path):
+        for file in files:
+            if file.endswith(('.hh', '.cc')):
+                out.append(Path(root) / file)
+    out.sort()
+    return out
+
+
+class TestCheckFormatStrings(unittest.TestCase):
+    PASS = 0
+    FAIL = 1
+
+    def test_valid_cases(self):
+        test_options = [
+            'format("{}", arg1)',
+            'format("{}{}=\"{}\"", RoutineName, s_ipsc->cCurrentModuleObject, s_ipsc->cAlphaArgs(1))',
+            'fmt::format("PLR          = {:7." + std::to_string(DecimalPrecision) + "F}", fmt::join(PLRArray, ","))',
+            'format("...{} is < 2 {{C}}. Freezing could occur.", cNumericFields(17))',
+            'format(R"({}="{}" invalid {}="{}" not found.)",\n CurrentModuleObject,\n ventSlab.Name,\n cAlphaFields(4),\n state.dataIPShortCut->cAlphaArgs(4))',
+            'format("{}{}{}{}{}{}", "Occurs for Node=", NodeName, ", ObjectType=", ObjectType, ", ObjectName=", ObjectName)',
+            'format("{}{}", RoutineName, "Node registered for both Parent and \"not\" Parent")'
+            'format("{}\n", EnergyPlus::Constant::unitNames[(int)meter->units])'
+        ]
+
+        tmp_dir = Path(tempfile.mkdtemp())
+        tmp_file = tmp_dir / "valid.cc"
+
+        with open(tmp_file, 'w') as f:
+            for line in test_options:
+                f.write(line)
+
+        self.assertEqual(self.PASS, check_format_statements([tmp_file]))
+
+    def test_invalid_cases(self):
+        test_options = [
+            'format("{}", arg1, arg2)',
+            'format("{}{}=\"{}\"")',
+        ]
+
+        tmp_dir = Path(tempfile.mkdtemp())
+        tmp_file = tmp_dir / "invalid.cc"
+
+        with open(tmp_file, 'w') as f:
+            for line in test_options:
+                f.write(line)
+
+        self.assertEqual(self.FAIL, check_format_statements([tmp_file]))
```

Comment:
There's a bunch of parsing voodoo in this script. These tests seem to be enough to capture most of our `format` uses.